File: Phase1/src/test/java/FileReaderTest.java
import org.junit.jupiter.api.Test;

import java.util.HashMap;

import static org.junit.jupiter.api.Assertions.assertEquals;

class FileReaderTest {
    @Test
    public void readNotAvailableFolder() {
        FileReader fileReader = new FileReader();
        HashMap<String,String> contents = fileReader.readingFiles("notAvailable");
        HashMap<String,String> expected = new HashMap<>();
        assertEquals(expected,contents);
    }

    @Test
    public void readSimpleFolder(){
        FileReader fileReader = new FileReader();
        HashMap<String,String> content = fileReader.readingFiles("src/test/simpleFolder");
        HashMap<String,String> expected = new HashMap<>();
        expected.put("simpleFile1.txt","this is simpleFile1");
        expected.put("simpleFile2.txt","this is simpleFile2");
        assertEquals(expected,content);
    }

    @Test
    public void readComplexFolder(){
        FileReader fileReader = new FileReader();
        HashMap<String,String> content = fileReader.readingFiles("src/test/complexFolder");
        HashMap<String,String> expected = new HashMap<>();
        expected.put("childFile1.txt","this is childFile1");
        expected.put("childFile2.txt","this is childFile2");
        expected.put("complexFile1.txt","this is complexFile1");
        expected.put("complexFile2.txt","this is complexFile2");
        assertEquals(expected,content);
    }
}

File: Phase5/SearchEngine/TestProject1/FileReaderTest.cs
using System;
using System.Collections.Generic;
using SearchEngine;
using Xunit;

namespace TestProject1
{
    public class FileReaderTest
    {
        [Fact]
        public void ReadNotAvailableFolder() {
            FileReader fileReader = new FileReader();
            Dictionary<string, string> contents = fileReader.ReadingFiles("notAvailable");
            Dictionary<string, string> expected = new Dictionary<string, string>();
            Assert.Equal(contents, expected);
        }

        [Fact]
        public void readSimpleFolder(){
            FileReader fileReader = new FileReader();
            Dictionary<string, string> content = fileReader.ReadingFiles("../../../TestProject1/TestDocs/simpleFolder");
            Dictionary<string, string> expected = new Dictionary<string, string>();
            expected.Add("simpleFile1.txt","this is simpleFile1");
            expected.Add("simpleFile2.txt","this is simpleFile2");
            Assert.Equal(expected,content);
        }

        [Fact]
        public void readComplexFolder(){
            FileReader fileReader = new FileReader();
            Dictionary<string, string> content = fileReader.ReadingFiles("../../../TestProject1/TestDocs/complexFolder");
            Dictionary<string, string> expected = new Dictionary<string, string>();
            expected.Add("childFile1.txt","this is childFile1");
            expected.Add("childFile2.txt","this is childFile2");
            expected.Add("complexFile1.txt","this is complexFile1");
            expected.Add("complexFile2.txt","this is complexFile2");
            Assert.Equal(expected,content);
        }


        [Fact]
        public void gettingFileAndReadIt (){
            FileReader fileReader = new FileReader();
            Dictionary<string, string> files = fileReader.ReadingFiles("../../../TestProject1/TestDocs/DocsForTest");

            Assert.True(files.ContainsKey("firstFile"));
            Assert.True(files["firstFile"].StartsWith("Hello Everyone"));
        }
    }
}